matrix-sdk/src/main/java/org/matrix/androidsdk/fragments/MatrixMessageListFragment.java

```diff
@@ -479,13 +481,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         if (null == mEventTimeLine) {
             mEventId = args.getString(ARG_EVENT_ID);
 
+            final String previewMode = args.getString(ARG_PREVIEW_MODE_ID);
             // the fragment displays the history around a message
             if (!TextUtils.isEmpty(mEventId)) {
                 mEventTimeLine = new EventTimeline(mSession.getDataHandler(), roomId, mEventId);
                 mRoom = mEventTimeLine.getRoom();
+                if (PREVIEW_MODE_UNREAD_MESSAGE.equals(previewMode)){
+                    mAdapter.setIsUnreadViewMode(true);
+                    final View footerView = ((LayoutInflater) getActivity().getSystemService(Context.LAYOUT_INFLATER_SERVICE))
+                            .inflate(R.layout.unread_messages_footer, null, false);
+                    mMessageListView.addFooterView(footerView);
```

Comment:
Don't you think it should be done on the application side ?
the unread mode adds only a footer to the listview.
It seems better that the activity / fragment decides to customise the UI itself.
a single Preview mode seems enough